packages/react-native-renderer/src/__tests__/ReactFabric-test.internal.js

```diff
@@ -220,6 +220,60 @@ describe('ReactFabric', () => {
     ).toMatchSnapshot();
   });
 
+  it.each([true, false])(
+    'should not clone nodes without children when updating props (feature flag = %p)',
+    async passChildren => {
```

Comment:
Because you set up a the feature flag with `__VARIANT__`, CI will already run these tests in both versions. (You can do this yourself on the command line by passing `--no-variant` to `yarn test`.)

You can then check the value of the flag with `gate(flags => flags.passChildrenWhenCloningPersistedNodes)`.

We prefer that pattern instead of mutating the ReactFeatureFlags module because it preserves the ability to run tests on the compiled build artifacts.

Reply:
Hmm, I'm unable to get this to work. It's only set as `__VARIANT__` in ReactFeatureFlags.native-fb-dynamic.js and as far as I can trace from logs, I only get the `false` path. Running with `yarn test -r www-modern ReactFabric-test` gets the variant, but the tests fail in other codepaths then.

Could there be a problem with this test explicitly mocking FeatureFlags itself?

```
jest.mock('shared/ReactFeatureFlags', () =>
  require('shared/forks/ReactFeatureFlags.native-oss'),
);
```

Reply:
Test failures were legit :) Bit weird to use www-modern for this, but it does give better coverage across tests.

Reply:
Oh that’s right I forgot we’ve never set up a test config for the Meta RN builds, like we did for www. I’ll work on adding one of those, not too difficult.

Reply:
https://github.com/facebook/react/pull/27489